astropy/utils/masked/function_helpers.py

```diff
@@ -655,13 +674,29 @@ def _masked_quantile_1d(a, q, **kwargs):
         return a.from_unmasked(np.zeros_like(a.unmasked, shape=q.shape), True)
 
 
-def _masked_quantile(a, q, axis=None, out=None, **kwargs):
+def _masked_quantile(a, q, axis, out, overwrite_input, method, *, _keepdims):
     # As for np.nanmedian, but without a fast option as yet.
+    # Note: np.quantile and np.lib.function_base._ureduce both have a `keepdims``
+    # argument. We preserve their distinction by taking a `_keepdims` argument here.
```

Comment:
It feels likely that the level of complexity here can be reduced, as it took me a couple iterations to get *a* working implementation. I'll revisit in a couple hours and see if I can refactor !